src/ripple/ripple.js

```diff
@@ -109,25 +110,52 @@
       }
       this.setRippleXY(x, y);
       this.setRippleStyles(true);
-      requestAnimationFrame(this.animFrameHandler.bind(this));
+
+      if (this.timeout) {
+        window.clearTimeout(this.timeout);
+      }
+
+      // We've just clicked, clear past upHandler_ call record
+      this.rippleElement_.mouseUp = false;
+      // Indicate to upHandler_ that ripple hasn't run for this.Constant_.DURATION
+      this.rippleElement_.animationCompleted = false;
+
+      window.requestAnimationFrame(function() {
```

Comment:
`requestAnimationFrame` changed to `window.requestAnimationFrame` for no reason.